Modix.Bot/Modules/MarkerRoleModule.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using Modix.Services.Core;
+using Modix.Data.Models.Core;
+
+namespace Modix.Modules
+{
+    [Group("pingrole"), Name("Marker Role Manager"), Summary("Allows you to add and remove specific marker roles.")]
+    public class MarkerRoleModule : ModuleBase
+    {
+        private readonly IAuthorizationService _authorizationService;
+        private readonly IDesignatedRoleService _designatedRoleService;
+
+        public MarkerRoleModule(IAuthorizationService authorizationService, IDesignatedRoleService designatedRoleService)
+        {
+            _authorizationService = authorizationService;
+            _designatedRoleService = designatedRoleService;
+        }
+
+        [Command("register")]
+        public async Task Register(IRole targetRole)
+        {
+            if(!(Context.User is IGuildUser user))
+            {
+                return;
+            }
+
+            if (user.RoleIds.Any(x => x == targetRole.Id))
+            {
+                await ReplyAsync("You're already registered to that role.");
+                return;
+            }
+
+            if (!await _designatedRoleService.RoleHasDesignationAsync(Context.Guild.Id, targetRole.Id, DesignatedRoleType.Pingable))
+            {
+                await ReplyAsync("Can't register to a role that isn't pingable.");
+                return;
+            }
+
+            await user.AddRoleAsync(targetRole);
+            await ReplyAsync($"Registered {user.Mention} to {Format.Bold(targetRole.Name)}.");
+        }
+
+        [Command("unregister")]
+        public async Task Unregister(IRole targetRole)
+        {
+            if (!(Context.User is IGuildUser user))
+            {
+                return;
+            }
+
+            if (!user.RoleIds.Any(x => x == targetRole.Id))
+            {
+                await ReplyAsync("You're not registered to that role.");
+                return;
+            }
+
+            if (!await _designatedRoleService.RoleHasDesignationAsync(Context.Guild.Id, targetRole.Id, DesignatedRoleType.Pingable))
+            {
+                await ReplyAsync("Can't unregister from a role that isn't pingable.");
+                return;
+            }
+
+            await user.RemoveRoleAsync(targetRole);
+            await ReplyAsync($"Unregistered {user.Mention} from role {Format.Bold(targetRole.Name)}.");
+        }
+
+        [Command("create")]
+        public async Task CreateRole(string targetRoleName)
```

Comment:
Protip (a quirk with commands): stick a `[Remainder]` attribute on this, or you'll have to escape roles with spaces with quotes when running the command.

Reply:
I did not know that, fixed.